File: archive/2025-11-old-scripts/fix-footer-canvas.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Fix Footer Canvas Template Setting
Enables the "Enable Layout for Elementor Canvas Template" option for footer template ID 73
"""

import requests
from requests.auth import HTTPBasicAuth
import json
import sys
import io

# Fix encoding for Windows console
if sys.platform == 'win32':
    sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8')

# Configuration
BASE_URL = "http://svetlinkielementor.local"
AUTH = HTTPBasicAuth("test", "test")
FOOTER_TEMPLATE_ID = 73

def enable_canvas_template():
    """Enable Canvas template option for footer"""

    print("[FIX] Fixing Footer Canvas Template Setting...\n")

    # Step 1: Get current template data
    print(f"[GET] Fetching footer template (ID {FOOTER_TEMPLATE_ID})...")
    response = requests.get(
        f"{BASE_URL}/wp-json/wp/v2/elementor_library/{FOOTER_TEMPLATE_ID}",
        auth=AUTH
    )

    if response.status_code != 200:
        print(f"[ERROR] Error fetching template: {response.status_code}")
        print(response.text)
        return False

    template = response.json()
    print(f"[OK] Template found: {template.get('title', {}).get('rendered', 'Untitled')}")

    # Step 2: Update meta to enable Canvas template
    print("\n[UPDATE] Enabling Canvas template option...")

    # The meta field is 'ehf-canvas-template' and should be set to 'enabled'
    update_data = {
        "meta": {
            "ehf-canvas-template": "enabled"
        }
    }

    response = requests.post(
        f"{BASE_URL}/wp-json/wp/v2/elementor_library/{FOOTER_TEMPLATE_ID}",
        auth=AUTH,
        json=update_data,
        headers={"Content-Type": "application/json"}
    )

    if response.status_code not in [200, 201]:
        print(f"[ERROR] Error updating template: {response.status_code}")
        print(response.text)
        return False

    print("[OK] Canvas template option enabled!")

    return True

def change_page_templates():
    """Change About, Programs, Contact, FAQ pages to Canvas template"""

    print("\n[UPDATE] Changing page templates to Canvas...\n")

    pages_to_update = [
        {"id": 23, "name": "About"},
        {"id": 25, "name": "Programs"},
        {"id": 27, "name": "Contact"},
        {"id": 29, "name": "FAQ"}
    ]

    for page in pages_to_update:
        print(f"[PAGE] Updating {page['name']} (ID {page['id']})...")

        # Update page template to Canvas
        update_data = {
            "meta": {
                "_wp_page_template": "elementor_canvas"
            }
        }

        response = requests.post(
            f"{BASE_URL}/wp-json/wp/v2/pages/{page['id']}",
            auth=AUTH,
            json=update_data,
            headers={"Content-Type": "application/json"}
        )

        if response.status_code in [200, 201]:
            print(f"  [OK] {page['name']} changed to Canvas template")
        else:
            print(f"  [WARN] Could not update {page['name']}: {response.status_code}")

    return True

def clear_elementor_cache():
    """Clear Elementor cache"""

    print("\n[CACHE] Clearing Elementor cache...")

    # Trigger cache clear by updating a dummy option
    # This will force Elementor to regenerate CSS
    response = requests.post(
        f"{BASE_URL}/wp-json/wp/v2/elementor_library/{FOOTER_TEMPLATE_ID}",
        auth=AUTH,
        json={"meta": {"_elementor_css": ""}},
        headers={"Content-Type": "application/json"}
    )

    print("[OK] Cache clear triggered")

    return True

def main():
    print("=" * 60)
    print("  FOOTER CANVAS TEMPLATE FIX")
    print("=" * 60)
    print()

    # Step 1: Enable Canvas template for footer
    if not enable_canvas_template():
        print("\n[ERROR] Failed to enable Canvas template")
        return

    # Step 2: Change page templates to Canvas
    if not change_page_templates():
        print("\n[WARN] Some pages could not be updated")

    # Step 3: Clear cache
    clear_elementor_cache()

    print("\n" + "=" * 60)
    print("[SUCCESS] FIX COMPLETE!")
    print("=" * 60)
    print()
    print("What was done:")
    print("  1. [OK] Enabled Canvas template option on Footer (ID 73)")
    print("  2. [OK] Changed About/Programs/Contact/FAQ to Canvas template")
    print("  3. [OK] Cleared Elementor cache")
    print()
    print("Next steps:")
    print("  1. Visit http://svetlinkielementor.local/ - Footer should now appear")
    print("  2. Visit About/Programs/Contact/FAQ - Footer should appear")
    print("  3. If still not showing, go to Elementor > Tools > Regenerate Files")
    print()

if __name__ == "__main__":
    main()

File: app/public/fix-footer-canvas.php
<?php
/**
 * Fix Footer Canvas Template Setting
 * Run this file directly: http://svetlinkielementor.local/fix-footer-canvas.php
 */

// Load WordPress
require_once(__DIR__ . '/wp-load.php');

// Security check
$auth_key = isset($_GET['key']) ? $_GET['key'] : '';
if ($auth_key !== 'fix_footer_canvas_2025') {
    die('Unauthorized. Add ?key=fix_footer_canvas_2025 to URL');
}

echo "<pre>\n";
echo "============================================================\n";
echo "  FOOTER CANVAS TEMPLATE FIX\n";
echo "============================================================\n\n";

// Step 1: Enable Canvas template option for Footer (ID 73)
echo "[FIX] Enabling Canvas template option for Footer (ID 73)...\n";

$footer_id = 73;

// Update the meta field
$result = update_post_meta($footer_id, 'ehf-canvas-template', 'enabled');

if ($result !== false) {
    echo "[OK] Canvas template option enabled!\n";
} else {
    $current = get_post_meta($footer_id, 'ehf-canvas-template', true);
    if ($current === 'enabled') {
        echo "[INFO] Canvas template option was already enabled\n";
    } else {
        echo "[ERROR] Failed to enable Canvas template option\n";
    }
}

// Verify the setting
$canvas_enabled = get_post_meta($footer_id, 'ehf-canvas-template', true);
echo "[VERIFY] Current value: " . ($canvas_enabled ? $canvas_enabled : 'not set') . "\n\n";

// Step 2: Change page templates to Canvas
echo "[UPDATE] Changing page templates to Canvas...\n\n";

$pages_to_update = [
    ['id' => 23, 'name' => 'About'],
    ['id' => 25, 'name' => 'Programs'],
    ['id' => 27, 'name' => 'Contact'],
    ['id' => 29, 'name' => 'FAQ']
];

foreach ($pages_to_update as $page) {
    echo "[PAGE] Updating {$page['name']} (ID {$page['id']})...\n";

    $result = update_post_meta($page['id'], '_wp_page_template', 'elementor_canvas');

    if ($result !== false) {
        echo "  [OK] {$page['name']} changed to Canvas template\n";
    } else {
        $current = get_post_meta($page['id'], '_wp_page_template', true);
        if ($current === 'elementor_canvas') {
            echo "  [INFO] {$page['name']} was already using Canvas template\n";
        } else {
            echo "  [WARN] Could not update {$page['name']}\n";
        }
    }
}

// Step 3: Clear Elementor cache
echo "\n[CACHE] Clearing Elementor cache...\n";

// Clear CSS cache
update_post_meta($footer_id, '_elementor_css', '');
foreach ($pages_to_update as $page) {
    update_post_meta($page['id'], '_elementor_css', '');
}

// Clear Elementor file cache if available
if (class_exists('\Elementor\Plugin')) {
    \Elementor\Plugin::$instance->files_manager->clear_cache();
    echo "[OK] Elementor cache cleared\n";
} else {
    echo "[WARN] Elementor plugin not found, cache not cleared\n";
}

echo "\n============================================================\n";
echo "[SUCCESS] FIX COMPLETE!\n";
echo "============================================================\n\n";

echo "What was done:\n";
echo "  1. [OK] Enabled Canvas template option on Footer (ID 73)\n";
echo "  2. [OK] Changed About/Programs/Contact/FAQ to Canvas template\n";
echo "  3. [OK] Cleared Elementor cache\n\n";

echo "Next steps:\n";
echo "  1. Visit http://svetlinkielementor.local/ - Footer should now appear\n";
echo "  2. Visit About/Programs/Contact/FAQ - Footer should appear\n";
echo "  3. Go to Elementor > Tools > Regenerate Files for full cache clear\n\n";

echo "============================================================\n";
echo "</pre>";
?>
